src/quark-core/src/QuarkStateManager.sol

```diff
@@ -9,76 +9,47 @@ import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
  * @author Compound Labs, Inc.
  */
 contract QuarkStateManager {
-    error NonceAlreadySet();
-    error NoUnusedNonces();
+    error NonReplayableNonce(address wallet, bytes32 nonce, bytes32 replayToken);
+    error InvalidReplayToken(address wallet, bytes32 nonce, bytes32 replayToken);
 
-    /// @notice Bit-packed nonce values
-    mapping(address wallet => mapping(uint256 bucket => uint256 bitset)) public nonces;
+    event NonceSubmitted(address wallet, bytes32 nonce, bytes32 replayToken);
 
-    /**
-     * @notice Return whether a nonce has been exhausted
-     * @param wallet Address of the wallet owning the nonce
-     * @param nonce Nonce to check
-     * @return Whether the nonce has been exhausted
-     */
-    function isNonceSet(address wallet, uint96 nonce) public view returns (bool) {
-        (uint256 bucket, uint256 mask) = getBucket(nonce);
-        return isNonceSetInternal(wallet, bucket, mask);
-    }
+    /// @notice Represents the unclaimed bytes32 value.
+    bytes32 public constant CLAIMABLE_TOKEN = bytes32(uint256(0));
```

Comment:
nit: `UNCLAIMED_TOKEN`?

Reply:
I posted a longer riff on naming in `#quark`, but I suggested:

`FREE` instead of `CLAIMABLE_TOKEN`, and `USED` instead of `NO_REPLAY_TOKEN`.

I also suggested renaming `nonceTokens` to `submitted`, so then it reads like `submitted[wallet][nonce] = FREE` or `USED` or whatever was the last-submitted token.

Finally, in this naming scheme `getNonceToken` would be `getSubmitted`.